source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/teamexplorer/pages/TeamExplorerHomePage.java

```diff
@@ -170,6 +189,135 @@ private void createConnectedUI(
         final TeamExplorerContext context) {
     }
 
+    private void createProtocolHandlerUI(
+        final FormToolkit toolkit,
+        final Composite parent,
+        final TeamExplorerContext context) {
+
+        if (!TeamExplorerHelpers.hasProtocolHandlerRequest()) {
+            return;
+        }
+
+        final Composite composite = toolkit.createComposite(parent);
+        composite.setBackground(TeamExplorerHelpers.getDropCompositeBackground(parent));
+        SWTUtil.gridLayout(composite, 1, false, 3, 3);
+        GridDataBuilder.newInstance().hAlignFill().hGrab().vIndent(5).applyTo(composite);
+
+        final Composite innerComposite = toolkit.createComposite(composite);
+        innerComposite.setBackground(parent.getDisplay().getSystemColor(SWT.COLOR_INFO_BACKGROUND));
+        innerComposite.setForeground(parent.getDisplay().getSystemColor(SWT.COLOR_INFO_FOREGROUND));
+        SWTUtil.gridLayout(innerComposite, 2, false, 5, 5);
+        GridDataBuilder.newInstance().hAlignFill().hGrab().applyTo(innerComposite);
+
+        final TeamExplorerNavigationLinkConfig[] navHomPageLinks = configuration.getNavigationLinks(HOME_PAGE_ID);
+        for (final TeamExplorerNavigationLinkConfig item : navHomPageLinks) {
+            final ITeamExplorerNavigationLink navLink = item.createInstance();
+            if (navLink.isVisible(context)) {
+                final Composite itemComposite = toolkit.createComposite(innerComposite);
+                itemComposite.setBackground(innerComposite.getBackground());
+                SWTUtil.gridLayout(itemComposite, 1, false, 5, 5);
+                GridDataBuilder.newInstance().hAlignFill().hGrab().applyTo(itemComposite);
+
+                final String encodedRepoUrl = TeamExplorerHelpers.getProtocolHandlerEncodedUrl();
+                final StringBuilder sb = new StringBuilder(encodedRepoUrl);
```

Comment:
URI builder requires JAX-RS library. I prefer not to add it for just a single use.